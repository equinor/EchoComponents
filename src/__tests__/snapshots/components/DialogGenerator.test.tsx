import React from 'react';
import renderer from 'react-test-renderer';
import { DialogGenerator } from '../../..';

it('renders full DialogGenerator', () => {
    const tree = renderer
        .create(
            <DialogGenerator
                dialogStyle={{ width: '300px' }}
                title={'This is the title'}
                actionButtons={[
                    { title: 'I changed my mind', onClick: jest.fn() },
                    { title: 'I agree', onClick: jest.fn(), variant: 'outlined' }
                ]}
            >
                <p>Hello from the inside</p>
            </DialogGenerator>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders DialogGenerator without buttons', () => {
    const tree = renderer
        .create(
            <DialogGenerator title={'This is the title'} actionButtons={[]}>
                <p>Hello from the inside</p>
            </DialogGenerator>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
