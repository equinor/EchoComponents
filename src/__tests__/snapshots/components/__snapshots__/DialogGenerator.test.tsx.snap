// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DialogGenerator without buttons 1`] = `
<div
  aria-describedby="eds-dialog-customcontent"
  aria-labelledby="eds-dialog-title"
  aria-modal={true}
  className="sc-cBNfnY gvVtXL"
  role="dialog"
  tabIndex={0}
>
  <div
    className="sc-jcVebW gncZGd"
    id="eds-dialog-title"
  >
    This is the title
  </div>
  <hr
    className="sc-jrAGrp duBmmQ sc-bZSQDF QRFwg"
  />
  <div
    className="sc-iUuytg cHAQWi"
    id="eds-dialog-customcontent"
  >
    <p>
      Hello from the inside
    </p>
  </div>
</div>
`;

exports[`renders full DialogGenerator 1`] = `
<div
  aria-describedby="eds-dialog-customcontent"
  aria-labelledby="eds-dialog-title"
  aria-modal={true}
  className="sc-cBNfnY gvVtXL"
  role="dialog"
  style={
    Object {
      "width": "300px",
    }
  }
  tabIndex={0}
>
  <div
    className="sc-jcVebW gncZGd"
    id="eds-dialog-title"
  >
    This is the title
  </div>
  <hr
    className="sc-jrAGrp duBmmQ sc-bZSQDF QRFwg"
  />
  <div
    className="sc-iUuytg cHAQWi"
    id="eds-dialog-customcontent"
  >
    <p>
      Hello from the inside
    </p>
  </div>
  <div
    className="sc-citwmv jgLzeC footer"
  >
    <button
      className="sc-gsTCUz gLPsnZ"
      disabled={false}
      onClick={[MockFunction]}
      tabIndex={0}
      type="button"
    >
      <span
        className="sc-bdfBwQ ibGDsh"
      >
        I changed my mind
      </span>
    </button>
    <button
      className="sc-gsTCUz ROJxf"
      disabled={false}
      onClick={[MockFunction]}
      tabIndex={0}
      type="button"
    >
      <span
        className="sc-bdfBwQ ibGDsh"
      >
        I agree
      </span>
    </button>
  </div>
</div>
`;
